### DISCUSSION TITLE
Typing clipboard content

### DISCUSSION THREAD

**Discussion by westerlind** [rustdesk/rustdesk#5451] [Q&A] [answered]
I am missing the ability to automatically type the content of my local clipboard on the remote computer I'm connected to.

This is a feature that is available in ScreenConnect (and probably other applications too), and it is extremely useful for:
1. Solving the issue of typing an admin password into a client-computer without leaking the password to the user via their clipboard (this assumes that you also disable the clipboard syncing while doing this, of course)
2. Being able to enter a really complex password fast on a client computer where pasting is not possible in the password field (for example UAC prompts on Windows)

There could for example be an option for this in one of the menus that open from the toolbar, and it would type in the content of your local clipboard on the remote computer instead of pasting it, and that's it.

**Comment by Affection4036** [comment]
This would be a VERY important change for me as I use the pro license to control several servers. I have to manually type passwords now while most of them are generated 50 character passwords. This would save me a lot of time!

**Comment by rustdesk** [comment]
How was "Typing clipboard content" called in ScreenConnect? could you share the screenshot?

**Reply by westerlind** [reply]
It's called "Send Clipboard Keystrokes", at least in the Mac version, but I guess it's called the same on Windows.

This is where you find it on the Mac version of the ScreenConnect client (I think it's a button in the toolbar on Windows, but I'm not 100% sure that I remember that correctly)

<img width="491" alt="image" src="https://github.com/user-attachments/assets/e151ca5f-826a-4e8a-a42f-1db4c645cc1b">

**Comment by rustdesk** [comment, marked as answer]
https://github.com/rustdesk/rustdesk/commit/bbdce8d57bcebe07be87c822afe50a5d7b518fda

**Comment by westerlind** [comment]
Thanks!

I compiled the master from GitHub to try this out, and I can confirm that it works as intended for me, which is great!

Can also add, for anyone who wants to know, that only the client needs to be updated for this to work. I have the same version as before on the rustdesk-server and on the machine I connected to.

<img width="291" alt="image" src="https://github.com/user-attachments/assets/91fd5031-2048-476c-900e-60773a6de50f">

**Reply by mahescho** [reply]
Thanks! This is most important for Windows login screens wehre no cut&paste is possible!
